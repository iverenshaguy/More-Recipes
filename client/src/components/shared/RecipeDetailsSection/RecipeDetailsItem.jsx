import React from 'react';
import PropTypes from 'prop-types';

/**
 * @exports
 * Returns Recipe Details Item
 * @param {object} props
 * @returns {JSX} RecipeDetailsItem
 */
const RecipeDetailsItem = ({ section }) => (
  <div className="p-xs-1 px-md-5 pt-3">
    <div className="row">
      <div className="col-xs-12 col-md-4 col-lg-4">
        <h4 className="uppercase">{section.title}</h4>
      </div>
      <div className="col-xs-12 col-md-8 col-lg-8 mr-0 pb-1" id="recipe-details-info">
        {section.info}
      </div>
    </div>
    <hr />
    <div className="text-justify" id={section.id}>
      {section.items.length === 0 &&
        <p className="mb-3">There are no {section.title.toLowerCase()}.</p>}
      {section.items &&
        section.items.map(item => <p className="mb-3" key={item}> {item}</p>)}
    </div>
  </div>
);

RecipeDetailsItem.propTypes = {
  section: PropTypes.shape({
    items: PropTypes.array.isRequired,
    title: PropTypes.string.isRequired,
    id: PropTypes.string.isRequired,
    info: PropTypes.element
  }).isRequired
};

export default RecipeDetailsItem;
