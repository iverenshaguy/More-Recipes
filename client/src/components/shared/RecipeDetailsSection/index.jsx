import React, { Fragment } from 'react';
import RecipeDetailsItem from './RecipeDetailsItem';
import { singleRecipePropTypes } from '../../../helpers/proptypes';

/**
 * @exports
 * Returns Recipe Item
 * @param {object} recipe
 * @returns {JSX} RecipeDetailsSection
 */
const RecipeDetailsSection = ({ recipe }) => {
  const ingredientsSection = {
    items: recipe.ingredients,
    title: 'Ingredients',
    id: 'ingredient-list',
    info: (
      <Fragment>
        <p className="inline-p text-muted"><i className="aria-hidden fa fa-clock-o fa-lg" /> {recipe.totalTime}<span className="hidden-sm-down">  |</span></p>
        <p className="inline-p text-muted"><i className="aria-hidden flaticon flaticon-oven-kitchen-tool-for-cooking-foods" /> {recipe.difficulty}</p>
      </Fragment>
    )
  };

  const preparationSection = {
    items: recipe.preparations,
    title: 'Preparation',
    id: 'preparation',
    info: (
      <Fragment>
        <p className="inline-p text-muted"><i className="aria-hidden fa fa-clock-o fa-lg" /> {recipe.prepTime}</p>
      </Fragment>
    )
  };

  const cookingDirectionsSection = {
    items: recipe.directions,
    title: 'Cooking Directions',
    id: 'cooking-directions',
    info: (
      <Fragment>
        <p className="inline-p text-muted"><i className="aria-hidden fa fa-clock-o" /> {recipe.cookTime} <span className="hidden-sm-down"> |&nbsp;</span></p>
        <p className="inline-p text-muted"><i className="aria-hidden fa fa-clock-o" /> Ready - {recipe.totalTime}</p>
      </Fragment>
    )
  };
  return (
    <Fragment>
      <RecipeDetailsItem section={ingredientsSection} />
      <RecipeDetailsItem section={preparationSection} />
      <RecipeDetailsItem section={cookingDirectionsSection} />
    </Fragment>
  );
};

RecipeDetailsSection.propTypes = {
  ...singleRecipePropTypes
};

export default RecipeDetailsSection;
