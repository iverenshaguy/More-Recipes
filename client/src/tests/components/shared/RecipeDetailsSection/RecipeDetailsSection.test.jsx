import React from 'react';
import RecipeDetailsSection from '../../../../components/shared/RecipeDetailsSection';

const recipe = {
  id: 2,
  recipeName: 'Recipe',
  userId: 1,
  isReviewed: false,
  ingredients: ['rice', 'beans'],
  preparations: ['wash', 'cook'],
  directions: ['wash', 'cook'],
  User: {
    id: 1,
    username: 'name'
  }
};

describe('RecipeDetailsSection', () => {
  it('renders correctly', () => {
    const wrapper = mount(<RecipeDetailsSection recipe={recipe} />);
    expect(toJson(wrapper)).toMatchSnapshot();
    wrapper.unmount();
  });
});
