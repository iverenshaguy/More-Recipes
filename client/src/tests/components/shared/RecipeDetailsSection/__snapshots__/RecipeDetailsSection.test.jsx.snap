// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecipeDetailsSection renders correctly 1`] = `
<RecipeDetailsSection
  recipe={
    Object {
      "User": Object {
        "id": 1,
        "username": "name",
      },
      "directions": Array [
        "wash",
        "cook",
      ],
      "id": 2,
      "ingredients": Array [
        "rice",
        "beans",
      ],
      "isReviewed": false,
      "preparations": Array [
        "wash",
        "cook",
      ],
      "recipeName": "Recipe",
      "userId": 1,
    }
  }
>
  <RecipeDetailsItem
    section={
      Object {
        "id": "ingredient-list",
        "info": <UNDEFINED>
          <p
            className="inline-p text-muted"
          >
            <i
              className="aria-hidden fa fa-clock-o fa-lg"
            />
             
            <span
              className="hidden-sm-down"
            >
                |
            </span>
          </p>
          <p
            className="inline-p text-muted"
          >
            <i
              className="aria-hidden flaticon flaticon-oven-kitchen-tool-for-cooking-foods"
            />
             
          </p>
        </UNDEFINED>,
        "items": Array [
          "rice",
          "beans",
        ],
        "title": "Ingredients",
      }
    }
  >
    <div
      className="p-xs-1 px-md-5 pt-3"
    >
      <div
        className="row"
      >
        <div
          className="col-xs-12 col-md-4 col-lg-4"
        >
          <h4
            className="uppercase"
          >
            Ingredients
          </h4>
        </div>
        <div
          className="col-xs-12 col-md-8 col-lg-8 mr-0 pb-1"
          id="recipe-details-info"
        >
          <p
            className="inline-p text-muted"
          >
            <i
              className="aria-hidden fa fa-clock-o fa-lg"
            />
             
            <span
              className="hidden-sm-down"
            >
                |
            </span>
          </p>
          <p
            className="inline-p text-muted"
          >
            <i
              className="aria-hidden flaticon flaticon-oven-kitchen-tool-for-cooking-foods"
            />
             
          </p>
        </div>
      </div>
      <hr />
      <div
        className="text-justify"
        id="ingredient-list"
      >
        <p
          className="mb-3"
          key="rice"
        >
           
          rice
        </p>
        <p
          className="mb-3"
          key="beans"
        >
           
          beans
        </p>
      </div>
    </div>
  </RecipeDetailsItem>
  <RecipeDetailsItem
    section={
      Object {
        "id": "preparation",
        "info": <UNDEFINED>
          <p
            className="inline-p text-muted"
          >
            <i
              className="aria-hidden fa fa-clock-o fa-lg"
            />
             
          </p>
        </UNDEFINED>,
        "items": Array [
          "wash",
          "cook",
        ],
        "title": "Preparation",
      }
    }
  >
    <div
      className="p-xs-1 px-md-5 pt-3"
    >
      <div
        className="row"
      >
        <div
          className="col-xs-12 col-md-4 col-lg-4"
        >
          <h4
            className="uppercase"
          >
            Preparation
          </h4>
        </div>
        <div
          className="col-xs-12 col-md-8 col-lg-8 mr-0 pb-1"
          id="recipe-details-info"
        >
          <p
            className="inline-p text-muted"
          >
            <i
              className="aria-hidden fa fa-clock-o fa-lg"
            />
             
          </p>
        </div>
      </div>
      <hr />
      <div
        className="text-justify"
        id="preparation"
      >
        <p
          className="mb-3"
          key="wash"
        >
           
          wash
        </p>
        <p
          className="mb-3"
          key="cook"
        >
           
          cook
        </p>
      </div>
    </div>
  </RecipeDetailsItem>
  <RecipeDetailsItem
    section={
      Object {
        "id": "cooking-directions",
        "info": <UNDEFINED>
          <p
            className="inline-p text-muted"
          >
            <i
              className="aria-hidden fa fa-clock-o"
            />
             
             
            <span
              className="hidden-sm-down"
            >
               | 
            </span>
          </p>
          <p
            className="inline-p text-muted"
          >
            <i
              className="aria-hidden fa fa-clock-o"
            />
             Ready - 
          </p>
        </UNDEFINED>,
        "items": Array [
          "wash",
          "cook",
        ],
        "title": "Cooking Directions",
      }
    }
  >
    <div
      className="p-xs-1 px-md-5 pt-3"
    >
      <div
        className="row"
      >
        <div
          className="col-xs-12 col-md-4 col-lg-4"
        >
          <h4
            className="uppercase"
          >
            Cooking Directions
          </h4>
        </div>
        <div
          className="col-xs-12 col-md-8 col-lg-8 mr-0 pb-1"
          id="recipe-details-info"
        >
          <p
            className="inline-p text-muted"
          >
            <i
              className="aria-hidden fa fa-clock-o"
            />
             
             
            <span
              className="hidden-sm-down"
            >
               | 
            </span>
          </p>
          <p
            className="inline-p text-muted"
          >
            <i
              className="aria-hidden fa fa-clock-o"
            />
             Ready - 
          </p>
        </div>
      </div>
      <hr />
      <div
        className="text-justify"
        id="cooking-directions"
      >
        <p
          className="mb-3"
          key="wash"
        >
           
          wash
        </p>
        <p
          className="mb-3"
          key="cook"
        >
           
          cook
        </p>
      </div>
    </div>
  </RecipeDetailsItem>
</RecipeDetailsSection>
`;
